c10/core/Allocator.h

```diff
@@ -230,6 +231,22 @@ struct C10_API InefficientStdFunctionContext {
   std::function<void(void*)> deleter_;
   InefficientStdFunctionContext(void* ptr, std::function<void(void*)> deleter)
       : ptr_(ptr), deleter_(std::move(deleter)) {}
+  InefficientStdFunctionContext(const InefficientStdFunctionContext&) = delete;
+  InefficientStdFunctionContext(InefficientStdFunctionContext&& rhs) noexcept
+      : ptr_(rhs.ptr_), deleter_(std::move(rhs.deleter_)) {
+    rhs.ptr_ = nullptr;
+  }
+  InefficientStdFunctionContext& operator=(
+      const InefficientStdFunctionContext&) = delete;
+  // NOLINTNEXTLINE(performance-noexcept-move-constructor)
+  InefficientStdFunctionContext& operator=(
+      InefficientStdFunctionContext&& rhs) {
+    this->~InefficientStdFunctionContext();
+    ptr_ = rhs.ptr_;
```

Comment:
Checkout `std::exchange` for simplifying some of this this with setting nullptrs